mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/search/QuotaSearcher.java

```diff
@@ -0,0 +1,28 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.quota.mailing.search;
+
+import java.util.List;
+
+import org.apache.james.core.User;
+
+public interface QuotaSearcher {
+
+    List<User> search(QuotaQuery query, int limit, int offset);
```

Comment:
Shouldn't the limit and the offset be optional?

Reply:
Don't we have strong typed objects for these kind of concerns?

Reply:
do we really return a user? Or a quota?

Reply:
A user is enough IMO. That is enough for retrieving the future route payloads